server/sonar-server/src/main/java/org/sonar/server/computation/queue/InternalCeQueueImpl.java

```diff
@@ -101,9 +101,9 @@ public void remove(CeTask task, CeActivityDto.Status status, CeTaskResult taskRe
 
   private static void updateTaskResult(CeActivityDto activityDto, @Nullable CeTaskResult taskResult) {
     if (taskResult != null) {
-      Long snapshotId = taskResult.getSnapshotId();
-      if (snapshotId != null) {
-        activityDto.setSnapshotId(snapshotId);
+      java.util.Optional<String> analysisUuid = taskResult.getAnalysisUuid();
```

Comment:
the move to `java.util.Optional` should be discusses and organised, I think, it's generating ugliness in code :(